packages/core/src/helpers/createNodeFromContent.ts

```diff
@@ -46,18 +47,58 @@ export function createNodeFromContent(
 
       return schema.nodeFromJSON(content)
     } catch (error) {
+
+      if (options.errorOnInvalidContent) {
+        throw new Error('[tiptap error]: Invalid JSON content', { cause: error as Error })
+      }
+
       console.warn('[tiptap warn]: Invalid content.', 'Passed value:', content, 'Error:', error)
 
       return createNodeFromContent('', schema, options)
     }
   }
 
   if (isTextContent) {
-    const parser = DOMParser.fromSchema(schema)
+    let schemaToUse = schema
+    let hasInvalidContent = false
+
+    // Only ever check for invalid content if we're supposed to throw an error
+    if (options.errorOnInvalidContent) {
+      schemaToUse = new Schema({
+        topNode: schema.spec.topNode,
+        marks: schema.spec.marks,
+        // Prosemirror's schemas are executed such that: the last to execute, matches last
+        // This means that we can add a catch-all node at the end of the schema to catch any content that we don't know how to handle
+        nodes: schema.spec.nodes.append({
```

Comment:
It is enough to have this as the last node, as it also captures any unknown element, even marks, right?

Reply:
yea, from my understanding since this is trying to parse HTML any thing that does not match should match this. I can add a couple more test cases to verify this

Reply:
So I've looked further into this, yes it does catch marks but only those that aren't purely just a style rule. There does not seem to be anyway to capture those and this would not be detected by that.

So for example if you did not have the bold extension and `<p style="font-weight: bold">test</p>` came through, it would say that no error occurred and strip the content silently.

I tried to make a catch all StyleParseRule like this:
```
.append({
          __tiptap__private__unknown__catch__all__mark: {
            content: 'inline*',
            group: 'inline',
            parseDOM: [
              {
                style: '*',
                getAttrs: () => {
                  console.log('is called')
                  // If this is ever called, we know that the content has something that we don't know how to handle in the schema
                  hasInvalidContent = true
                  return null
                },
              },
            ],
          },
        })
```

But, [due to the way that prosemirror matches styles](https://github.com/ProseMirror/prosemirror-model/blob/d61616994c1907f6856aa2cf027a0e4944fc8023/src/from_dom.ts#L248), it has to be an exact match. If we wanted to be absolutely certain on this, then we would need to move upstream to prose-mirrors parser for this.

Reply:
I think having this on node level first is good. We can try to discuss with Marjin how we could solve this problem with Marks and come back to it later.